### PR TITLE
Engine async create

### PR DESCRIPTION
**Pull request by GuoLei1990**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features) https://github.com/ant-galaxy/oasis-engine.github.io/pull/679
- [x] Toolkit https://github.com/ant-galaxy/oasis-engine-toolkit/pull/174 

### What is the current behavior? (You can also link to an open issue here)
```ts
PhysXPhysics.initialize().then(() => {
  const engine = new WebGLEngine("canvas");
  engine.physicsManager.initialize(PhysXPhysics);
  ......
}
```

### What is the new behavior (if this is a feature change)?
```ts
WebGLEngine.create({ canvas: "canvas", physics: new PhysXPhysics()}).then(
  (engine) => {
  ......
  }
```

### CODE REVIEW

**Review by zhuxudong — APPROVED** [review]
+1

File: packages/core/src/Engine.ts [hunk 1]
```diff
@@ -518,6 +513,23 @@ export class Engine extends EventDispatcher {
     }
   }
 
+  protected _initialize(configuration: EngineConfiguration): Promise<Engine> {
+    return new Promise((resolve, reject) => {
+      const physics = configuration.physics;
+      if (physics) {
+        physics
```
Comment (gz65555) [inline packages/core/src/Engine.ts:520]: 这些 `initialize` 可以优化一下

Comment (gz65555) [reply packages/core/src/Engine.ts:520]: 通过一个注册方法注册
```typescript
export function registerInitialize<T>(
  initializeFunction: (config: T) => void
): void {
  // Do nothing.
  initializeFunctions.push(initializeFunction);
}
```

Comment (gz65555) [reply packages/core/src/Engine.ts:520]: 通过 declare module 可以拓展 config
```typescript
declare module "./config" {
  interface IConfiguration {
    /**
     * KTX2 loader configuration.
     */
    ktx2Loader: IKTX2Loader;
  }
}
```

**Pull request merged**